column_buffer.go

```diff
@@ -900,6 +900,95 @@ func (col *booleanColumnBuffer) ReadValuesAt(values []Value, offset int64) (n in
 	}
 }
 
+type int8ColumnBuffer struct{ int32Page }
+
+func newInt8ColumnBuffer(typ Type, columnIndex int16, numValues int32) *int8ColumnBuffer {
+	return &int8ColumnBuffer{
+		int32Page: int32Page{
+			typ:         typ,
+			values:      make([]int32, 0, numValues),
+			columnIndex: ^columnIndex,
+		},
+	}
+}
+
+func (col *int8ColumnBuffer) Clone() ColumnBuffer {
+	return &int8ColumnBuffer{
+		int32Page: int32Page{
+			typ:         col.typ,
+			values:      append([]int32{}, col.values...),
+			columnIndex: col.columnIndex,
+		},
+	}
+}
+
+func (col *int8ColumnBuffer) ColumnIndex() ColumnIndex { return int8ColumnIndex{&col.int32Page} }
+
+func (col *int8ColumnBuffer) OffsetIndex() OffsetIndex { return int8OffsetIndex{&col.int32Page} }
+
+func (col *int8ColumnBuffer) BloomFilter() BloomFilter { return nil }
+
+func (col *int8ColumnBuffer) Dictionary() Dictionary { return nil }
+
+func (col *int8ColumnBuffer) Pages() Pages { return onePage(col.Page()) }
+
+func (col *int8ColumnBuffer) Page() Page { return &col.int32Page }
+
+func (col *int8ColumnBuffer) Reset() { col.values = col.values[:0] }
+
+func (col *int8ColumnBuffer) Cap() int { return cap(col.values) }
+
+func (col *int8ColumnBuffer) Len() int { return len(col.values) }
+
+func (col *int8ColumnBuffer) Less(i, j int) bool { return col.values[i] < col.values[j] }
+
+func (col *int8ColumnBuffer) Swap(i, j int) {
+	col.values[i], col.values[j] = col.values[j], col.values[i]
+}
+
+func (col *int8ColumnBuffer) Write(b []byte) (int, error) {
```

Comment:
It's not really clear to me when this gets used, so I'm not sure if it's expected for each byte to map to a separate int8 or for four bytes to map to an int32. I think this code assumes the latter, but it was copied from the int32 class, so I may misunderstand it.